core/unit_test/incremental/Test12a_ThreadScratch.hpp

```diff
@@ -0,0 +1,101 @@
+/*
+//@HEADER
+// ************************************************************************
+//
+// Kokkos v. 2.0
+// Copyright (2014) Sandia Corporation
+//
+// Under the terms of Contract DE-AC04-94AL85000 with Sandia Corporation,
+// the U.S. Government retains certain rights in this software.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+// 1. Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright
+// notice, this list of conditions and the following disclaimer in the
+// documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY SANDIA CORPORATION "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL SANDIA CORPORATION OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Christian R. Trott (crtrott@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+*/
+
+// @Kokkos_Feature_Level_Required:6
+// Unit test for hierarchical parallelism
+// Create concurrent work hierarchically and verify if
+// contributions of paticipating processing units corresponds to expected value
+// Use a scratch pad memory for each team
+#include <gtest/gtest.h>
+#include <Kokkos_Core.hpp>
+
+// Degrees of concurrency per nesting level
+#define N 16
+#define M 16
+
+namespace Test {
+
+template <class ExecSpace>
+struct ThreadScratch {
+  void run() {
+    typedef Kokkos::TeamPolicy<ExecSpace> teamPolicy;
+    typedef typename Kokkos::TeamPolicy<ExecSpace>::member_type memberType;
+    typedef Kokkos::View<int **, ExecSpace> dataType;
+    typedef Kokkos::View<int *, ExecSpace,
+                         Kokkos::MemoryTraits<Kokkos::Unmanaged> >
+        scratchDataType;
```

Comment:
Kokkos uses PascalCase (or sometimes snake_case) for types and camelCase for variables.  Please change.  Also prefer a type alias to a `typedef`.

Reply:
This has not been addressed